src/main/java/uk/gov/companieshouse/api/accounts/validation/TangibleAssetsValidator.java

```diff
@@ -0,0 +1,931 @@
+        }
+    }
+
+    private void validateCurrentNetBookValue(Errors errors, TangibleAssetsResource tangibleAssetsResource, TangibleSubResource subResource) {
+
+        Long costAtPeriodEnd =
+                getCostAtPeriodEnd(tangibleAssetsResource);
+
+        Long depreciationAtPeriodEnd =
+                getDepreciationAtPeriodEnd(tangibleAssetsResource);
+
+        Long calculatedCurrentNetBookValue = costAtPeriodEnd - depreciationAtPeriodEnd;
+
+        if (!tangibleAssetsResource.getNetBookValueAtEndOfCurrentPeriod().equals(calculatedCurrentNetBookValue)) {
+
+            addError(errors, incorrectTotal, getJsonPath(subResource, NET_BOOK_VALUE_CURRENT_PERIOD));
+        }
+    }
+
+    private void validatePreviousNetBookValue(Errors errors, TangibleAssetsResource tangibleAssetsResource, TangibleSubResource subResource) {
+
+        Long costAtPeriodStart =
+                getCostAtPeriodStart(tangibleAssetsResource);
+
+        Long depreciationAtPeriodStart =
+                getDepreciationAtPeriodStart(tangibleAssetsResource);
+
+        Long calculatedPreviousNetBookValue = costAtPeriodStart - depreciationAtPeriodStart;
+
+        if (!tangibleAssetsResource.getNetBookValueAtEndOfPreviousPeriod().equals(calculatedPreviousNetBookValue)) {
+
+            addError(errors, incorrectTotal, getJsonPath(subResource, NET_BOOK_VALUE_PREVIOUS_PERIOD));
+        }
+    }
+
+    private void validatePresenceOfMultipleYearFilerFields(Errors errors, TangibleAssetsResource tangibleAssetsResource, TangibleSubResource subResource, List<TangibleSubResource> invalidSubResources) {
+
+        boolean subResourceInvalid = false;
+
+        if (tangibleAssetsResource.getNetBookValueAtEndOfCurrentPeriod() != null  || tangibleAssetsResource.getNetBookValueAtEndOfPreviousPeriod() != null) {
+
+            if (tangibleAssetsResource.getNetBookValueAtEndOfPreviousPeriod() == null) {
+
+                addError(errors, valueRequired, getJsonPath(subResource, NET_BOOK_VALUE_PREVIOUS_PERIOD));
+                subResourceInvalid = true;
+            }
+
+            if (tangibleAssetsResource.getNetBookValueAtEndOfCurrentPeriod() == null) {
+
+                addError(errors, valueRequired, getJsonPath(subResource, NET_BOOK_VALUE_CURRENT_PERIOD));
+                subResourceInvalid = true;
+            }
+
+            if (tangibleAssetsResource.getCost() == null || tangibleAssetsResource.getCost().getAtPeriodStart() == null) {
+
+                addError(errors, valueRequired, getJsonPath(subResource, COST_AT_PERIOD_START));
+                subResourceInvalid = true;
+            }
+
+            if (tangibleAssetsResource.getCost() == null || tangibleAssetsResource.getCost().getAtPeriodEnd() == null) {
+
+                addError(errors, valueRequired, getJsonPath(subResource, COST_AT_PERIOD_END));
+                subResourceInvalid = true;
+            }
+
+            if (tangibleAssetsResource.getDepreciation() != null) {
```

Comment:
Not 100% sure on the validation logic require - but could we merge these two if statements to be 

` if ((tangibleAssetsResource.getDepreciation() != null) && (hasDepreciationFieldsSet(tangibleAssetsResource))) {
`

Reply:
Multiple if statements have been compounded into 1